math/linear_recurrence_matrix.cpp

```diff
@@ -0,0 +1,368 @@
+/**
+ * @brief Evaluate recurrence relation using matrix exponentiation.
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂